Sources/Git/Utils.swift

```diff
@@ -0,0 +1,66 @@
+import clibgit
+
+final class Box<T> {
+    let value: T
+    init(_ value: T) {
+        self.value = value
+    }
+}
+
+public enum BranchType {
+    case local
+    case remote
+    case all
+
+    var git_branch_t: git_branch_t {
+        switch self {
+        case .local:
+            return GIT_BRANCH_LOCAL
+        case .remote:
+            return GIT_BRANCH_REMOTE
+        case .all:
+            return GIT_BRANCH_ALL
+        }
+    }
+}
+
+private var isInitialized = false
+
+@discardableResult
+func wrap(_ closure: () throws -> Int32) throws -> git_error_code {
+    if !isInitialized {
+        try _wrap { git_libgit2_init() }
+    }
+
+    return try _wrap { try closure() }
+}
```

Comment:
It feels kinda annoying to me to litter code with wrap.... don't almost all operations start with some kind of repository open or such where we could centralize this?